WcaOnRails/app/models/light_result.rb

```diff
@@ -32,34 +33,26 @@ def initialize(r)
     @best = r["best"]
     @average = r["average"]
     @personName = r["personName"]
-    @eventId = r["eventId"]
-    @formatId = r["formatId"]
-    @roundId = r["roundId"]
     @pos = r["pos"]
     @personId = r["personId"]
     @regionalSingleRecord = r["regionalSingleRecord"]
     @regionalAverageRecord = r["regionalAverageRecord"]
-    @countryId = r["countryId"]
+    @country = Country.find(r["countryId"])
+    @format = Format.find(r["formatId"])
+    @round = Round.find(r["roundId"])
+    @event = Event.find(r["eventId"])
```

Comment:
Could you maybe fetch country, format, round and event before constructing the class and pass them in as arguments? I feel very uneasy about side-effects (mutating global state + network request) in constructors. I should be able to call `Foo.new` several times without worrying about this stuff.